src/v/kafka/security/sasl_authentication.h

```diff
@@ -59,6 +60,8 @@ class sasl_server final {
         _mechanism = std::move(m);
     }
 
+    const ss::sstring& principal() const { return _mechanism->principal(); }
```

Comment:
should `principal` be a named type? `using acl_principal = ...`

Reply:
probably, but not in this patch. this patch adds acls

Reply:
https://github.com/vectorizedio/redpanda/issues/988